amazon-cloudwatch-agent.yaml

```diff
@@ -24,25 +25,38 @@ pipeline:
     with:
       deps: github.com/opencontainers/runc@v1.1.14
 
-  - runs: |
-      make amazon-cloudwatch-agent-linux
-      mkdir -p ${{targets.contextdir}}/usr/bin
-      # upstream linux debs and rpms include this in their respective package.
-      # the upstream container image doesn't include this.
-      cp packaging/opentelemetry-jmx-metrics.jar ${{targets.contextdir}}/usr/bin/
-      cp -r build/bin/linux_$(go env GOARCH)/* ${{targets.contextdir}}/usr/bin/
+  - uses: go/build
+    with:
+      packages: ./cmd/amazon-cloudwatch-agent
+      ldflags: |
+        -X github.com/aws/amazon-cloudwatch-agent/cfg/agentinfo.VersionStr=${{package.version}}
+        -X github.com/aws/amazon-cloudwatch-agent/cfg/agentinfo.BuildStr=$(date -u +"%Y-%m-%dT%H:%M:%SZ")
+      output: amazon-cloudwatch-agent
 
 subpackages:
+  - range: amazon-cloudwatch-agent-packages
+    name: amazon-cloudwatch-agent-${{range.key}}
+    pipeline:
+      - uses: go/build
+        with:
+          packages: ./cmd/${{range.key}}
+          ldflags: |
+            -X github.com/aws/amazon-cloudwatch-agent/cfg/agentinfo.VersionStr=${{package.version}}
+            -X github.com/aws/amazon-cloudwatch-agent/cfg/agentinfo.BuildStr=$(date -u +"%Y-%m-%dT%H:%M:%SZ")
+          output: ${{range.key}}
+
   - name: ${{package.name}}-compat
-    description: "Compatibility package for locating binaries according to upstream Dockerfile"
+    description: "upstream image have executable placed at /"
     pipeline:
       - runs: |
           mkdir -p "${{targets.contextdir}}"/opt/aws/amazon-cloudwatch-agent/bin
-          ln -s usr/bin/linux_$(go env GOARCH)/amazon-cloudwatch-agent "${{targets.contextdir}}"/opt/aws/amazon-cloudwatch-agent/bin/amazon-cloudwatch-agent
-          ln -s usr/bin/linux_$(go env GOARCH)/amazon-cloudwatch-agent-config-wizard "${{targets.contextdir}}"/opt/aws/amazon-cloudwatch-agent/bin/amazon-cloudwatch-agent-wizard
-          ln -s usr/bin/linux_$(go env GOARCH)/config-downloader "${{targets.contextdir}}"/opt/aws/amazon-cloudwatch-agent/bin/config-downloader
-          ln -s usr/bin/linux_$(go env GOARCH)/config-translator "${{targets.contextdir}}"/opt/aws/amazon-cloudwatch-agent/bin/config-translator
-          ln -s usr/bin/linux_$(go env GOARCH)/start-amazon-cloudwatch-agent "${{targets.contextdir}}"/opt/aws/amazon-cloudwatch-agent/bin/start-amazon-cloudwatch-agent
+          mkdir -p "${{targets.contextdir}}"/opt/aws/amazon-cloudwatch-agent/etc
```

Comment:
can't you just use range here too?

Reply:
I avoided that because that'll give us individual `-compat` packages. I thought we could combine all the packages and use one compat one. (thought this would be easier when importing compat package for fips image). 

Please let me know if you think that would a better appraoch, I'll range over it.

Reply:
makes sense, yeah, thanks!